crates/astria-grpc-mock-test/tests/health/main.rs

```diff
@@ -3,270 +3,7 @@
-    let server = start_mock_server().await;
-    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
-        .await
-        .unwrap();
-    let expected_request = HealthCheckRequest {
-        service: "helloworld".to_string(),
-    };
-    let expected_response = HealthCheckResponse {
-        status: 1,
-    };
-    let mock = Mock::for_rpc_given("check", matcher::message_exact_pbjson(&expected_request))
-        .respond_with(response::constant_response(expected_response.clone()));
-    server.mocked.register(mock).await;
-    let rsp = client
-        .check(HealthCheckRequest {
-            service: "helloworld".to_string(),
-        })
-        .await
-        .unwrap();
-    assert_eq!(&expected_response, rsp.get_ref());
-}
-
-#[tokio::test]
-async fn partial_pbjson_match_works() {
-    let server = start_mock_server().await;
-    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
-        .await
-        .unwrap();
-    let expected_request = HealthCheckRequest {
-        service: "helloworld".to_string(),
-    };
-    let expected_response = HealthCheckResponse {
-        status: 1,
-    };
-    // FIXME: Right now this is equivalent to an exact check because the request only has one field.
-    let mock = Mock::for_rpc_given("check", matcher::message_partial_pbjson(&expected_request))
-        .respond_with(response::constant_response(expected_response.clone()));
-    server.mocked.register(mock).await;
-    let rsp = client
-        .check(HealthCheckRequest {
-            service: "helloworld".to_string(),
-        })
-        .await
-        .unwrap();
-    assert_eq!(&expected_response, rsp.get_ref());
-}
-
-#[tokio::test]
-#[should_panic]
-async fn incorrect_mock_response_fails_server() {
-    let server = start_mock_server().await;
-    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
-        .await
-        .unwrap();
-    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
-        .respond_with(response::default_response::<HealthCheckRequest>());
-    server.mocked.register(mock).await;
-    let _ = client
-        .check(HealthCheckRequest {
-            service: "helloworld".to_string(),
-        })
-        .await;
-}
-
-#[tokio::test]
-#[should_panic]
-async fn incorrect_mock_response_fails_guard() {
-    let server = start_mock_server().await;
-    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
-        .await
-        .unwrap();
-    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
-        .respond_with(response::default_response::<HealthCheckRequest>());
-
-    let guard = server.mocked.register_as_scoped(mock).await;
-    let check = client.check(HealthCheckRequest {
-        service: "helloworld".to_string(),
-    });
-
-    let _ = join!(guard.wait_until_satisfied(), check);
-}
+mod test_matcher;
```

Comment:
This looks good, but please avoid prepending `test_` for every submodule.

Reply:
Done in 8a4b360.